tests/API/OVAL/unittests/test_glob_to_regex.xml

```diff
@@ -0,0 +1,351 @@
+      <glob_to_regex glob_noescape="false">
+        <literal_component datatype="string">*</literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:12" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="false">
+        <literal_component datatype="string">?</literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:13" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="false">
+        <literal_component datatype="string">\*</literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:14" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="false">
+        <literal_component datatype="string">\?</literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:15" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="false">
+        <literal_component datatype="string">\[hello\]</literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:16" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="false">
+        <literal_component datatype="string">x[[:digit:]]\*</literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:17" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="false">
+        <literal_component datatype="string"></literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:18" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="false">
+        <literal_component datatype="string">~/files/*.txt</literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:19" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="true">
+        <literal_component datatype="string">\*</literal_component>
+      </glob_to_regex>
+    </local_variable>
+    <local_variable id="oval:x:var:20" version="1" comment="Applying the glob_to_regex function to the glob string." datatype="string">
+      <glob_to_regex glob_noescape="true">
+        <literal_component datatype="string">\?</literal_component>
+      </glob_to_regex>
+    </local_variable>
```

Comment:
Please add the naughty invalid globs that cause buffer overrun in the _glob_to_regex function right now. It would be valuable to make sure we don't regress in the future.

Maybe add them as a separate definition.